### ISSUE TITLE
Emitting the entire bash completions script from my program?

### ISSUE THREAD

**Issue by wrouesnel** [kislyuk/argcomplete#248]
Over in the kingpin project for Go https://github.com/alecthomas/kingpin#bashzsh-shell-completion bash completions are made available from a program by a user simply specifying the binary with a special command line argument as part of their shell.

I like packaging all my scripts as single-file executables, so this is something I'd like to do with `argcomplete` but I can't seem to find a way to do it - argcomplete seems to depend on the completions script being installed on the user system in it's entirety.

Could an example be added which gives this functionality somehow, if it's possible?

**Comment by kislyuk** [comment]
Yes, this is possible using recent functionality from #237:
```
import argparse, argcomplete, sys
class ShellcodeAction(argparse.Action):
    def __call__(self, *args, **kwargs):
        print(argcomplete.shellcode(sys.argv[0]))
parser = argparse.ArgumentParser()
parser.add_argument("--print-completion-shellcode", action=ShellcodeAction, nargs=0)
argcomplete.autocomplete(parser)
parser.parse_args()
```
Please feel free to open a PR to add this to the docs.

**Comment by kislyuk** [comment]
P.S. I just released this (#237) in v1.9.4.
